### DISCUSSION TITLE
Module Migration Help

### DISCUSSION THREAD

**Discussion by mal45van4** [oqtane/oqtane.framework#3830] [Q&A] [answered]
Two questions for everyone, very new to this. When specifying the module versions in Module Info, you reference it as 1.0.1, however, if you have a migration script with version 01000101, or something similar, do you put 1.0.1.1 in the module info release versions in order to have it triggered?

Secondly, I noticed that in the migration cs scripts, there is a downgrade method. If I wanted to downgrade a module version back to 1.0.0, what is the appropriate way to have this triggered (i.e., dropping a column created in 1.0.1)?

Thanks!

**Comment by leigh-pointer** [comment, marked as answer]
Hi @mal45van4 
> Two questions for everyone, very new to this. When specifying the module versions in Module Info, you reference it as 1.0.1, however, if you have a migration script with version 01000101, or something similar, do you put 1.0.1.1 in the module info release versions in order to have it triggered?

You will need to add it to the ReleaseVersions in your ModuleInfo class
`ReleaseVersions = "1.0.0,1.0.1.1",` 

- where 1.0.1.1 is your new release.  You will also need to create a new migrations file in **[YourModule]\Server\Migrations\**
01000101_NameOFYourMirgration.cs (naming to your standards)  What is important is that this file is decorated with a Migration attribute.
```
{
    [DbContext(typeof(ModuleStationContext))]
    [Migration(Owner.Module.YourModule.01.00.01.01")]
    public class InitializeModule : MultiDatabaseMigration
    {
```

Question 2, I have never tried this.

**Reply by mal45van4** [reply]
Thanks, @leigh-pointer!
